File: MATLAB_model/python_scripts/SimulationSettings.py
import os

class SimulationSettings:
    def __init__(self, ShowPlots=0, SaveSim=0, SavePlotStuff=0, SavePop=0,
                 NormalCycle=1, LutStim=0, FollStim=0, DoubStim=0,
                 Foll_ModelPop=0, Horm_ModelPop=0, workDir=None, outputDir=None):
        # save simulation results
        self.showPlots = ShowPlots
        self.saveSim = SaveSim
        self.savePlotsStuff = SavePlotStuff
        self.savePop = SavePop
        # select type of simulation
        self.normalCycle = NormalCycle
        self.lut_stim = LutStim
        self.foll_stim = FollStim
        self.doub_stim = DoubStim
        self.foll_modelPop = Foll_ModelPop
        self.horm_modelPop = Horm_ModelPop
        # directories
        if not os.path.exists(workDir):
            print(f"Directory '{workDir}' doesn't exit. Value of workDir set to None")
            self.workDir = None
        else:
            self.workDir = workDir
        self.outputDir = outputDir
        if not os.path.exists(outputDir):
            os.makedirs(outputDir)
            print(f"Directory '{outputDir}' created.")
        else:
            print(f"Directory '{outputDir}' already exists.")

File: MATLAB_model/SimulationSettings.m
classdef SimulationSettings
    properties
        % saving settings
        showPlots
        saveSim
        savePlotStuff
        savePop
        % type of simulation
        stim
        simulationType
        % directories settings
        OutputDir
        FSHVecPath
        StartVecPath
    end
    methods
        function obj = SimulationSettings(simulationType)
            switch simulationType
                case 'LutStim'
                    obj.simulationType = SimulationType.LutSim;
                case 'FollStim'
                    obj.simulationType = SimulationType.FollStim;
                case 'DoubleStim'
                    obj.simulationType = SimulationType.DoubleStim;
                case 'FollModelPop'
                    obj.simulationType = SimulationType.FollModelPop;
                case 'HormModelPop'
                    obj.simulationType = SimulationType.HormModelPop;
                case 'NormalCycle'
                    obj.simulationType = SimulationType.NormalCycle;
                otherwise
                    obj.simulationType = SimulationType.NormalCycle;
            end
            obj.showPlots     = 1;
            obj.saveSim       = 1;
            obj.savePlotStuff = 1;
            obj.savePop       = 0;
            % type of simulation
            obj.stim          = 0;
            % directories settings
            obj.OutputDir = './hormone_populations';
            obj.FSHVecPath = './ModelPopulation/ControlRun/FSHS.txt';
            obj.StartVecPath = './ModelPopulation/ControlRun/StartTimesPoiss.txt';
        end
    end
end
